### ISSUE TITLE
Fix S1607 FN: Support derived attributes

### ISSUE THREAD

**Issue by gregory-paidis-sonarsource** [SonarSource/sonar-dotnet#6397]
### Description

FN for S1607 (TestMethodShouldNotBeIgnored).

Using C#11 generic attributes, the rule is not raising an issue.

### Repro steps

```csharp
    using Microsoft.VisualStudio.TestTools.UnitTesting;

    [DerivedTestClassAttribute<int>]
    class TestSuite
    {
        [DerivedTestMethodAttribute<int>]
        [Ignore] // FN
        public void Foo1()
        {
        }

        [DerivedDataTestMethodAttribute<int>]
        [Ignore] // FN
        public void Foo2()
        {
        }
    }

    public class DerivedTestClassAttribute<T> : TestClassAttribute { }

    public class DerivedTestMethodAttribute<T> : TestMethodAttribute { }

    public class DerivedDataTestMethodAttribute<T> : DataTestMethodAttribute { }
```

### Expected behavior

The rule should raise an issue.

### Actual behavior

The rule does not raise an issue.
